packages/search/src/search.tsx

```diff
@@ -90,349 +90,352 @@ type State = {
 	hasFocus: boolean;
 };
 
-export class Search extends React.Component< Props, State > {
-	static defaultProps = {
-		autoFocus: false,
-		compact: false,
-		delaySearch: false,
-		delayTimeout: SEARCH_DEBOUNCE_MS,
-		describedBy: undefined,
-		dir: undefined,
-		disableAutocorrect: false,
-		disabled: false,
-		fitsContainer: false,
-		hideClose: false,
-		hideOpenIcon: false,
-		defaultIsOpen: false,
-		onClick: noop,
-		onKeyDown: noop,
-		onSearchChange: noop,
-		onSearchOpen: noop,
-		onSearchClose: noop,
-		openIconSide: 'left' as const,
-		//undefined value for overlayStyling is an optimization that will
-		//disable overlay scrolling calculation when no overlay is provided.
-		overlayStyling: undefined,
-		pinned: false,
-		recordEvent: noop,
-		searching: false,
-	};
-
-	static currentId = 0;
-	static getUniqueId = (): number => {
-		return Search.currentId++;
-	};
-
-	instanceId = Search.getUniqueId();
-	searchInput = React.createRef< HTMLInputElement >();
-	openIcon = React.createRef< HTMLButtonElement >();
-	overlay = React.createRef< HTMLDivElement >();
-
-	// debounced `onSearch` will have a `cancel` function
-	onSearch: ( ( search: string ) => void ) & { cancel?: () => void } = this.props.delaySearch
-		? debounce( this.props.onSearch, this.props.delayTimeout )
-		: this.props.onSearch;
-
-	state = {
-		keyword: this.props.defaultValue ?? '',
-		isOpen: this.props.defaultIsOpen ?? false,
-		hasFocus: this.props.autoFocus ?? false,
-	};
-
-	openSearch = (
-		event:
-			| MouseEvent< HTMLButtonElement | HTMLInputElement >
-			| KeyboardEvent< HTMLButtonElement | HTMLInputElement >
-	): void => {
-		event.preventDefault();
-		this.setState( {
-			keyword: '',
-			isOpen: true,
-		} );
-		this.props.onSearchOpen?.( event );
-		// prevent outlines around the open icon after being clicked
-		this.openIcon.current?.blur();
-		this.props.recordEvent?.( 'Clicked Open Search' );
-	};
-
-	closeSearch = (
-		event:
-			| MouseEvent< HTMLButtonElement | HTMLInputElement >
-			| KeyboardEvent< HTMLButtonElement | HTMLInputElement >
-	): void => {
-		event.preventDefault();
-
-		if ( this.props.disabled ) {
-			return;
-		}
-
-		this.setState( {
-			keyword: '',
-			isOpen: false,
-		} );
-
-		if ( this.searchInput.current ) {
-			this.searchInput.current.value = ''; // will not trigger onChange
-		}
-
-		if ( this.props.pinned ) {
-			this.searchInput.current?.blur();
-			this.openIcon.current?.focus();
-		} else {
-			this.searchInput.current?.focus();
-		}
-
-		this.props.onSearchClose?.( event );
-
-		this.props.recordEvent?.( 'Clicked Close Search' );
-	};
-
-	closeListener = keyListener( this.closeSearch );
-	openListener = keyListener( this.openSearch );
+let currentId = 0;
+const getUniqueId = () => currentId++;
+
+//This is fix for IE11. Does not work on Edge.
```

Comment:
nit: comment spacing, but it's mostly weird that the lint check isn't asking about it 🤔